src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -324,33 +334,65 @@ private AnalyzerSettings GetLanguageSpecificSettings(AnalysisConfig config)
             return settings;
         }
 
+        /// <summary>
+        /// Merges and returns the supplied list of analyzer paths. In case of duplicate
+        /// SonarAnalyzers, the path from the sonarAnalyzerPaths list is used.
+        /// </summary>
+        private string[] MergeAnalyzersLists(IEnumerable<string> sonarAnalyzerPaths, IEnumerable<string> userProvidedAnalyzerPaths)
+        {
+            var nonNullSonarAnalyzerPaths = sonarAnalyzerPaths ?? Enumerable.Empty<string>();
+            var nonNullUserProvidedAnalyzerPaths = userProvidedAnalyzerPaths ?? Enumerable.Empty<string>();
+
+            var sonarAnalyzerDuplicates = nonNullUserProvidedAnalyzerPaths
+               .Where(x =>
+               {
+                   var userProvidedfileName = GetFileName(x);
+                   return sonarDotnetPluginNames.Any(pluginName => string.Equals(pluginName, userProvidedfileName, StringComparison.OrdinalIgnoreCase));
+               })
+               .ToArray();
```

Comment:
can this be made more LINQy?

also, given that `sonarDotnetPluginNames` is accessed multiple times, should it rather be an `ISet` - `new HashSet<string>(array, StringComparer.OrdinalIgnoreCase);` ? 

```suggestion
            var sonarAnalyzerDuplicates = nonNullUserProvidedAnalyzerPaths
               .Select(x => GetFileName(x))
               .Where(x =>sonarDotnetPluginNames.Contains(x))
               .ToArray();
```